### PR TITLE
Set PF_HTTP_CONNECTION_PROVIDER_ENDPOINT to env in retrieve_tool_func_result

### PR DESCRIPTION
**Pull request by y1811**
# Description

Now runtime put `PF_HTTP_CONNECTION_PROVIDER_ENDPOINT` in request.environment_variables, executor should set it in env to init a http connection provider. This is used for get connection in dynamic list scenario.

# All Promptflow Contribution checklist:
- [ ] **The pull request does not introduce [breaking changes].**
- [ ] **CHANGELOG is updated for new features, bug fixes or other significant changes.**
- [ ] **I have read the [contribution guidelines](../CONTRIBUTING.md).**
- [ ] **Create an issue and link to the pull request to get dedicated review from promptflow team. Learn more: [suggested workflow](../CONTRIBUTING.md#suggested-workflow).**

## General Guidelines and Best Practices
- [ ] Title of the pull request is clear and informative.
- [ ] There are a small number of commits, each of which have an informative message. This means that previously merged commits do not appear in the history of the PR. For more information on cleaning up the commits in your PR, [see this page](https://github.com/Azure/azure-powershell/blob/master/documentation/development-docs/cleaning-up-commits.md).

### Testing Guidelines
- [ ] Pull request includes test coverage for the included changes.

### CODE REVIEW

File: src/promptflow-core/promptflow/executor/_service/utils/process_utils.py [hunk 1]
```diff
@@ -81,15 +82,19 @@ async def invoke_sync_function_in_process(
                 ProcessManager().remove_process(run_id)
 
 
-def _execute_target_function(
+def _set_environment_variables_then_execute_target_function(
```
Comment (PeiwenGaoMS) [inline src/promptflow-core/promptflow/executor/_service/utils/process_utils.py:85]: Maybe there is no need to change the function name, just adding a parameter would be enough.

File: src/promptflow-core/promptflow/executor/_service/utils/process_utils.py [hunk 2]
```diff
@@ -81,15 +82,19 @@ async def invoke_sync_function_in_process(
...
     target_function: Callable,
     args: tuple,
     kwargs: dict,
     return_dict: dict,
     error_dict: dict,
     context_dict: dict,
+    environment_variables: dict,
 ):
     block_terminate_signal_to_parent()
+    if environment_variables:
```
Comment (PeiwenGaoMS) [inline src/promptflow-core/promptflow/executor/_service/utils/process_utils.py:95]: There is a util function `set_environment_variables` in `src\promptflow-core\promptflow\executor\_service\utils\service_utils.py`. You can use it.

**Review by PeiwenGaoMS — DISMISSED** [review]

**Review by PeiwenGaoMS — APPROVED** [review]

File: src/promptflow-core/promptflow/executor/_service/apis/tool.py [hunk 3]
```diff
@@ -27,7 +27,14 @@ async def retrieve_tool_func_result(request: RetrieveToolFuncResultRequest):
     from promptflow._core.tools_manager import retrieve_tool_func_result
 
     args = (request.func_call_scenario, request.func_path, request.func_kwargs, request.ws_triple)
-    return await invoke_sync_function_in_process(retrieve_tool_func_result, args=args, wait_timeout=SHORT_WAIT_TIMEOUT)
+    # To support dynamic list, runtime put PF_HTTP_CONNECTION_PROVIDER_ENDPOINT in request.environment_variables,
```
Comment (thy09) [inline src/promptflow-core/promptflow/executor/_service/apis/tool.py:30]: Add a test case for this?

**Review by Adrian999999999 — APPROVED** [review]

File: src/promptflow-core/promptflow/executor/_service/apis/tool.py (hunk 3 above)
Comment (y1811) [reply src/promptflow-core/promptflow/executor/_service/apis/tool.py:30]: Thanks, added a ut to test invoke_sync_function_in_process with passing environment_variables and assert if it set the env.

**Review by PeiwenGaoMS — APPROVED** [review]

**Pull request merged**